### ISSUE TITLE
New Rule: shorter object syntax in arrow function (ES6)

### ISSUE THREAD

**Issue by hzoo** [jscs-dev/node-jscs#1385]
No idea what the name for this is, but can have both rules.

```js
// bad
  [1, 2, 3].map((x) => {
    return { number: x };
  });

// good
  [1, 2, 3].map((x) => ({ number: x });
```
an example from http://tc39wiki.calculist.org/es6/arrow-functions/

```js
// Parenthesize the body to return an object literal expression
let key_maker = val => ({key: val});
```

**Comment by alawatthe** [comment]
Do you only want to disallow the return statement when returning an object or in all single expression arrow functions? I.e something like

```js
// bad
x => {return 42;}

// good
x => 42
```

Name suggestion for the second case: disallowReturnInSingleExpressionArrowFunctions Does this win the price of the longest rule name so far? ;-)

**Comment by mrjoelkemp** [comment]
(require|disallow)ImplicitReturn or the same with InArrowFunction for
specificity (if necessary)?

**Comment by hzoo** [comment]
Related to #1340 - `requireShorthandArrowFunctions`

```js
// good
evens.map(v => v + 1);
// bad
evens.map(v => { return v + 1; });
```

This is specifically for returning a single object. 

Actually maybe we should just merge it into a single rule.

**Comment by ljharb** [comment]
Whatever you end up with, I'd want to be able to configure rules so that I could:
 - allow implicit return of non-object-literals
 - disallow implicit return of paren-wrapped object literals
 - disallow implicit return of blocks with labelled statements

and, potentially, disallow implicit return entirely, or, allow implicit return entirely. (in case that helps make "picking a rule name" harder :-) )

**Comment by hzoo** [comment]
Could some of those options could become their own rules? Or only apply within arrow functions?

Maybe we should have a `disallowLabelStatments` rule.. - (had to look at up haha).

**Comment by ljharb** [comment]
We absolutely should. Those should only be used intentionally imo, and that's what linters should catch :-)

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
